src/otx/core/data/dataset/action.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for OTXActionDataset."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable
+
+import cv2
+import torch
+from datumaro import Label
+
+if TYPE_CHECKING:
+    import numpy as np
+    from datumaro.components.dataset_base import DatasetItem
+
+from otx.core.data.dataset.base import OTXDataset
+from otx.core.data.entity.action import ActionClsBatchDataEntity, ActionClsDataEntity
+from otx.core.data.entity.base import ImageInfo
+
+
+class OTXActionClsDataset(OTXDataset[ActionClsDataEntity]):
+    """OTXDataset class for action classification task."""
+
+    def _get_item_impl(self, idx: int) -> ActionClsDataEntity | None:
+        item = self.dm_subset.get(id=self.ids[idx], subset=self.dm_subset.name)
+        # video_data = [frame.data for frame in video]
+        video_data = self._get_video_data(item)
```

Comment:
It seems you are grabbing whole frames from the video. This can be a problem if the video has a long sequence.

Reply:
Done by 93aed91